h2/src/main/org/h2/message/TraceObject.java

```diff
@@ -371,15 +371,15 @@ protected SQLException logAndConvert(Throwable ex) {
                     trace.error(e, "exception");
                 }
             }
-        } catch(Throwable ignore) {
+        } catch(Throwable another) {
             if (e == null) {
                 try {
                     e = new SQLException("GeneralError", "HY000", ErrorCode.GENERAL_ERROR_1, ex);
-                } catch (OutOfMemoryError ignored) {
-                    return DbException.SQL_OOME;
+                } catch (OutOfMemoryError | NoClassDefFoundError ignored) {
+                    e = DbException.SQL_OOME;
                 }
             }
-            e.addSuppressed(ignore);
+            e.addSuppressed(another);
```

Comment:
With these changes `e` can be `DbException.SQL_OOME`, I don't think that it's a good idea to accumulate suppressed exceptions in this constant.